src/Microsoft.Azure.WebJobs.Extensions.EventHubs/Listeners/EventHubListener.cs

```diff
@@ -71,6 +72,8 @@ void IListener.Cancel()
 
         void IDisposable.Dispose()
         {
+            _disposed = true;
+            StopAsync(CancellationToken.None).Wait();

```

Comment:
I think we should implement `IAsyncDisposable` here as well since this is an async operation. https://docs.microsoft.com/en-us/dotnet/standard/garbage-collection/implementing-disposeasync#implement-both-dispose-and-async-dispose-patterns

Unfortunately, I don't think functions will call `DisposeAsync` today but it'll be an improvement when it does.

Reply:
netstandard2.0 is target framework for EH ext track1 and `IAsyncDisposable` is not supported.

Reply:
ah! got it!

Reply:
FYI - you can pull in IAsyncDisposable via https://www.nuget.org/packages/Microsoft.Bcl.AsyncInterfaces/